libflux/src/flux/semantic/builtins.rs

```diff
@@ -0,0 +1,338 @@
+                "gen" => Node::Package(maplit::hashmap! {
+                    "tables" => Node::Builtin("forall [t0] (n: int, tags: [{name: string | cardinality: int}]) -> {_time: time | _value: float | t0}"),
+                }),
+                "promql" => Node::Package(maplit::hashmap! {
+                    "changes" => Node::Builtin("forall [t0, t1] (<-tables: [{_value: float | t0}]) -> [{_value: float | t1}]"),
+                    "promqlDayOfMonth" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "promqlDayOfWeek" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "promqlDaysInMonth" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "emptyTable" => Node::Builtin("forall [] () -> [{_start: time | _stop: time | _time: time | _value: float}]"),
+                    "extrapolatedRate" => Node::Builtin("forall [t0, t1] (<-tables: [{_start: time | _stop: time | _time: time | _value: float | t0}], ?isCounter: bool, ?isRate: bool) -> [{_value: float | t1}]"),
+                    "holtWinters" => Node::Builtin("forall [t0, t1] (<-tables: [{_time: time | _value: float | t0}], ?smoothingFactor: float, ?trendFactor: float) -> [{_value: float | t1}]"),
+                    "promqlHour" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "instantRate" => Node::Builtin("forall [t0, t1] (<-tables: [{_time: time | _value: float | t0}], ?isRate: bool) -> [{_value: float | t1}]"),
+                    "labelReplace" => Node::Builtin("forall [t0, t1] (<-tables: [{_value: float | t0}], source: string, destination: string, regex: string, replacement: string) -> [{_value: float | t1}]"),
+                    "linearRegression" => Node::Builtin("forall [t0, t1] (<-tables: [{_time: time | _stop: time | _value: float | t0}], ?predict: bool, ?fromNow: float) -> [{_value: float | t1}]"),
+                    "promqlMinute" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "promqlMonth" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "promHistogramQuantile" => Node::Builtin("forall [t0, t1] (<-tables: [t0], ?quantile: float, ?countColumn: string, ?upperBoundColumn: string, ?valueColumn: string) -> [t1]"),
+                    "resets" => Node::Builtin("forall [t0, t1] (<-tables: [{_value: float | t0}]) -> [{_value: float | t1}]"),
+                    "timestamp" => Node::Builtin("forall [t0] (<-tables: [{_value: float | t0}]) -> [{_value: float | t0}]"),
+                    "promqlYear" => Node::Builtin("forall [] (timestamp: float) -> float"),
+                    "join" => Node::Builtin("forall [t0, t1, t2] (left: [t0], right: [t1], fn: (left: t0, right: t1) -> t2) -> [t2]"),
+                }),
+            }),
+            "json" => Node::Package(maplit::hashmap! {
+                "encode" => Node::Builtin("forall [t0] (v: t0) -> bytes"),
```

Comment:
Does json.encode work on any input type or just records?

Reply:
It appears to work for any input type.